shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -37,19 +38,91 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
         public static string GetTypeDisplayName(Type type, bool fullName = true)
         {
             var sb = new StringBuilder();
-            ProcessTypeName(type, sb, fullName);
+            ProcessTypeName(type, sb, fullName ? NameFormatting.Full : NameFormatting.Short);
             return sb.ToString();
         }
 
-        private static void AppendGenericArguments(Type[] args, int startIndex, int numberOfArgsToAppend, StringBuilder sb, bool fullName)
+        public static string GetMethodDisplayName(MethodBase method, bool fullTypeName = true)
```

Comment:
This does not belong to TypeNameHelper. At least it should be put in separate file. I am not sure about the structure of this repo.

cc: @Eilon - To find if we need new project or name change or anything.

Reply:
I understand that method display, not an initial intent of `TypeNameHelper` but method display relay heavily on type display and I don't see a nice option of splitting this into separate cases.

Posible splitting options: 
- Exposing required method of `TypeNameHelper` by making them internal will make them visible for everybody who uses `TypeNameHelper`, because it's included in projects as internal class.
- Using only public methods of `TypeNameHelper` will result in a lot of `string` allocation and creation or redundant `StringBuilders`, and beside that existing method lack of flexibility in displaying open generics, adding new public methods will make class usage more complicated.

In some way method is part of type, added method won't break existing usages and allow display method names where needed (like list of constructors in DependencyInjection project).

Reply:
The way I understand is.
`DeclaringType.Method(parameterType parameterName,....)`

Yes, it is true that to print out type it would allocate many string builders. But apart from that I don't see any reason to use internal code of TypeNameHelper.
In other words, we can expose a method of TypeNameHelper which would take stringBuilder and append the type info to it. That would save allocations.

Reply:
Hmm not sure. I don't know the best place for this to live. Maybe in DI itself because that's the only place it's needed? And the code in DI can just use the `TypeNameHelper` that's here as a utility?

Reply:
I'll investigate option of moving this functionality to separate class, and to DI repository, but I think it should be near `TypeNameHelper`, because next time when somebody will need it (in Logger for example, or for some error message in another project) they will write it again.

If we look only at current situation in DI, method display could be used in DI and DI.Abstraction that will require duplicating functionality or making it public or making separate code package that depends on `TypeNameHelper`. Sure leaving it in the same package with `TypeNameHelper` not required, but helps a lot.